src/plugins/XHRUpload.js

```diff
@@ -90,6 +91,19 @@ module.exports = class XHRUpload extends Plugin {
       formPost.append(item, file.meta[item])
     })
 
+    let data = opts.data
+    if (typeof opts.data === 'function') {
+      data = opts.data(file)
+    }
+
+    if (data != null) {
+      (typeof data === 'object')
+        ? Object.keys(data).forEach((key) => {
+          formPost.append(key, data[key])
+        })
+        : formPost.append('data', data)
```

Comment:
I think we should just require that `data` is an object, and not allow single values.